src/app/peak-edit/peak-edit.component.ts

```diff
@@ -0,0 +1,752 @@
+        hwm_locationdescription: h.hwm_locationdescription,
+        hwm_notes: h.hwm_notes,
+        hwm_uncertainty: h.hwm_uncertainty,
+        uncertainty: h.uncertainty,
+        hwm_quality_id: h.hwm_quality_id,
+        hwm_type_id: h.hwm_type_id,
+        latitude_dd: h.latitude_dd,
+        longitude_dd: h.longitude_dd,
+        marker_id: h.marker_id,
+        peak_summary_id: h.peak_summary_id,
+        site_id: h.site_id,
+        stillwater: h.stillwater = "No" ? 0 : 1,
+        survey_date: h.survey_date,
+        survey_member_id: h.survey_member_id,
+        vcollect_method_id: h.vcollect_method_id,
+        vdatum_id: h.vdatum_id,
+        waterbody: h.waterbody,
+      };
+      return fhwm;
+  }
+
+  //add or remove a hwm from the list of chosen hwms for determining this peak
+  addHWM(hwm) {
+    let formattedHWM = this.formatHWM(hwm);
+    if (hwm.selected) {
+      // Only values not already selected initially are added
+      if(hwm.peak_summary_id === undefined || hwm.peak_summary_id === null){
+        this.selectedHWMs.push(formattedHWM);
+      }
+      // Check if has been re-added and remove from remove list
+      let index = this.removedHWMs.map(function (hwm) { return hwm.hwm_id; }).indexOf(hwm.hwm_id)
+      if(index !== -1){
+        this.removedHWMs.splice(index);
+      }
+    } else {
+        // Need to store removed ones for PUT
+        if (hwm.peak_summary_id !== undefined) {
+            this.removedHWMs.push(formattedHWM);
+        }
+        // Remove from selected HWMs
+        if (this.selectedHWMs.length > 0) {
+            let index = this.selectedHWMs.map(function (hwm) { return hwm.hwm_id; }).indexOf(hwm.hwm_id);
+            this.selectedHWMs.splice(index, 1);
+
+        }
+    }
+  }
+
+  primaryHWM(hwm) {
+    const dialogRef = this.dialog.open(ConfirmComponent, {
+      data: {
+        title: "Set as Primary",
+        titleIcon: "",
+        message: "Are you sure you want to set this as the Primary HWM? Doing so will populate the Peak Date (not including time), Stage, Vertical Datum and Height Above Ground.",
+        confirmButtonText: "OK",
+        showCancelButton: true,
+      },
+    });
+    dialogRef.afterClosed().subscribe((result) => {
+      // Confirm returns true
+      if(result){
+        this.peak.peak_date = new Date(hwm.flag_date);
+        this.peak.peak_stage = hwm.elev_ft;
+        this.peak.vdatum_id = hwm.vdatum_id;
+        this.peak.height_above_gnd = hwm.height_above_gnd;
+        let index = this.hwms.indexOf(hwm);
+        this.hwms[index].selected = true;
+      }
+    })
+  }
+
+  primaryDataFile(file) {
+    const dialogRef = this.dialog.open(ConfirmComponent, {
+      data: {
+        title: "Set as Primary",
+        titleIcon: "",
+        message: "Are you sure you want to set this as the Primary Data file? (Coming soon: Script processing to populate the Peak date, time and time zone, Stage, Vertical Datum and Height above ground)",
```

Comment:
```suggestion
        message: "Are you sure you want to set this as the Primary Data file?",
```

Let's remove this -- there is a chance we will incorporate these processing scripts at some point, but not anytime super soon.